app/src/test/java/com/example/myweatherapp/mvp/presenter/MyWeatherAppPresenterTest.kt

```diff
@@ -0,0 +1,129 @@
+package com.example.myweatherapp.mvp.presenter
+
+import com.example.myweatherapp.adapter.ItemClicked
+import com.example.myweatherapp.data.entity.DaysWeather
+import com.example.myweatherapp.data.service.WeatherService
+import com.example.myweatherapp.mvp.model.MyWeatherAppDetailModel
+import com.example.myweatherapp.mvp.model.MyWeatherAppModel
+import com.example.myweatherapp.mvp.utils.Utils
+import com.example.myweatherapp.mvp.utils.Utils.CITY
+import com.example.myweatherapp.mvp.utils.Utils.DATE_ONE
+import com.example.myweatherapp.mvp.utils.Utils.NUMBER_ZERO
+import com.example.myweatherapp.mvp.utils.Utils.weatherList
+import com.example.myweatherapp.mvp.view.MyWeatherAppView
+import com.nhaarman.mockitokotlin2.mock
+import com.nhaarman.mockitokotlin2.verify
+import com.nhaarman.mockitokotlin2.whenever
+import io.reactivex.rxjava3.android.plugins.RxAndroidPlugins
+import io.reactivex.rxjava3.core.Observable
+import io.reactivex.rxjava3.core.Scheduler
+import io.reactivex.rxjava3.disposables.Disposable
+import io.reactivex.rxjava3.internal.schedulers.ExecutorScheduler
+import io.reactivex.rxjava3.observers.TestObserver
+import io.reactivex.rxjava3.plugins.RxJavaPlugins
+import org.junit.Assert
+import org.junit.Before
+import org.junit.BeforeClass
+import org.junit.Test
+import java.util.concurrent.Executor
+import java.util.concurrent.TimeUnit
+
+class MyWeatherAppPresenterTest {
+
+    private var service: WeatherService = mock()
+    private var model = MyWeatherAppModel(service)
+    private val view: MyWeatherAppView = mock()
+    private lateinit var modelDetail: MyWeatherAppDetailModel
+    private lateinit var presenter: MyWeatherAppPresenter
+    private val item: ItemClicked = mock()
+    private lateinit var testObserver: TestObserver<List<DaysWeather>>
+
+    @Before
+    fun setUp() {
+        presenter = MyWeatherAppPresenter(model, view)
+        model = MyWeatherAppModel(WeatherService())
```

Comment:
why do you do this if you declare it in line 34?

Reply:
Because if I don't I can't test the test in which the service is called. I am talking about the test: "get weather forecast from a service"